packages/klighd-core/src/sidebar/sidebar.tsx

```diff
@@ -141,4 +142,17 @@ export class Sidebar extends AbstractUIExtension {
             this.actionDispatcher.dispatch(ToggleSidebarPanelAction.create(currentPanelID, "hide"));
         });
     }
+
+    /**
+     * Register a mouse left handler that hides the content if the mouse leaves the sidebar.
+     */
+    private addMouseLeaveListener(containerElement: HTMLElement): void {
+        containerElement.addEventListener("mouseleave", (e) => {
+            const currentPanelID = this.sidebarPanelRegistry.currentPanelID;
+            if (currentPanelID) {
+                this.actionDispatcher.dispatch(ToggleSidebarPanelAction.create(currentPanelID, "hide"));
+            }
```

Comment:
Actions should only be dispatched if they do something, this should have a guard checking for the panel being pinned.